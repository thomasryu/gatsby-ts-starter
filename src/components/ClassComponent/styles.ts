import { css } from 'emotion'

export const componentButton = css`
  display: block;
  height: 40px;
  width: 100px;
  margin-bottom: 50px;
  background-color: red;
  border-radius: 5px;

  &:hover {
    opacity: 0.6;
  }
`
export const componentPressedButton = css`
  background-color: green;
`
