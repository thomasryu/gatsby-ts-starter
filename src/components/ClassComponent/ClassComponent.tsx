import * as React from 'react'
import * as styles from './styles'

interface IState {
  pressed: boolean
}

interface IProps {}

export class ClassComponent extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props)
    this.state = {
      pressed: false
    }
  }

  private handleButtonClick = () => {
    this.setState({
      pressed: true
    })
  }

  public render() {
    return (
      <button
        className={`${styles.componentButton} ${this.state.pressed ? styles.componentPressedButton : ''}`}
        onClick={this.handleButtonClick}
      />
    )
  }
}
