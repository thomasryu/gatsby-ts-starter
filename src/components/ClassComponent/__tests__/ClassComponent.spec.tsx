import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { ClassComponent } from '../ClassComponent'

Enzyme.configure({ adapter: new Adapter() })

const createClassComponent = () => {
  const testTarget = shallow(<ClassComponent />)
  return { testTarget }
}

it('renders the component as expected', () => {
  const setup = createClassComponent()
  expect(setup.testTarget).toMatchSnapshot()
})
