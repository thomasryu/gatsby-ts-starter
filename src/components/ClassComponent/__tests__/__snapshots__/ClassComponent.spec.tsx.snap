// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ClassComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-cbzqkj ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-cbzqkj ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
